pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -90,9 +93,21 @@ class LightningKokkos(LightningQubit):
         "Identity",
     }
 
-    def __init__(self, wires, *, sync=True, c_dtype=np.complex128, shots=None, batch_obs=False):
+    def __init__(
+        self,
+        wires,
+        *,
+        sync=True,
+        c_dtype=np.complex128,
+        shots=None,
+        batch_obs=False,
+        kokkos_args=None,
+    ):
         super().__init__(wires, c_dtype=c_dtype, shots=shots)
-        self._kokkos_state = _kokkos_dtype(self._state.dtype)(self._state)
+        if kokkos_args is None:
+            self._kokkos_state = _kokkos_dtype(self._state.dtype)(self._state)
+        else:
+            self._kokkos_state = _kokkos_dtype(self._state.dtype)(self._state, kokkos_args)
```

Comment:
The Kokkos doc says 
> If you set num_threads or num_numa to zero or less, Kokkos will try to determine default values if possible or otherwise set them to 1. [...] Note that the default value of each parameter is -1.
These are the ones most relevant for OpenMP and Threads, and `kokkos_args` should be an `InitArguments`. To make this more robust, there are several things we could do here. Add an `isinstance` check on `kokkos_args` and add a `TypeError` in the else statement. We could add constructors for `InitArguments` objects at the Python layer doing some validation. We could also add getters/setters to the binding at the C++ layer.

Reply:
Yep, all of the above sound like good ideas.

Agreed, for now allowing control of the threading for the THREADS & OPENMP backends should suffice.